File: POO/aula14/Construtor.java
class Pai
{
	public Pai()
	{
		System.out.println("ESTOU NO CONSTRUTOR VAZIO DE PAI");
	}
	public Pai(int x)
	{
		System.out.println("ESTOU NO CONSTRUTOR INT " + x +" DE PAI");
	}
}

class Filho extends Pai
{
	public Filho()
	{
		System.out.println("ESTOU NO CONSTRUTOR VAZIO DE FILHO");
	}

	public Filho(int x)
	{
		super(x);
		System.out.println("ESTOU NO CONSTRUTOR INT DE FILHO");
	}
}

public class Construtor
{
	public static void main(String[] args)
	{
		Filho obj = new Filho(666);
	}
}

File: POO/aula14/Construtor.cpp
#include <iostream>

class Pai
{
	public:
		Pai()
		{
			std::cout << "ESTOU NO CONSTRUTOR VAZIO DE PAI\n";
		}
		Pai(int x)
		{
			std::cout << "ESTOU NO CONSTRUTOR INT ("<< x << ") DE PAI\n";
		}
};

class Filho : public Pai
{
	public:
		Filho(int x):Pai(x)
		{
			std::cout << "ESTOU NO CONSTRUTOR INT DE FILHO\n";
		}
};

int main()
{
	Filho obj(666);
	return 0;
}
